packages/react/src/primitives/types/textField.ts

```diff
@@ -36,4 +36,6 @@ export interface TextFieldOptions extends FieldProps, FlexContainerStyleProps {
   type?: InputProps['type'];
 }
 
-export interface TextFieldProps extends TextFieldOptions, InputProps {}
+export interface TextFieldProps
+  extends TextFieldOptions,
+    Omit<InputProps, 'checked'> {}
```

Comment:
This is a good point, I think we might need to rethink the `InputProps` type, it might be doing too much. Maybe moving things out to `FieldProps` which are shared across all *Field components.